### PR TITLE
keys will be searched from next page if next link header exists

### PR DESCRIPTION
**Pull request by sophiaso**
## Description
Support query keys from config store if next link header exists.

### CODE REVIEW

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java [hunk 1]
```diff
@@ -61,4 +92,15 @@
             throw new IllegalStateException(LOAD_FAILURE_MSG, e);
         }
     }
+
+    private static String getNextLink(CloseableHttpResponse response) {
+        // Extract next link path and query from response header
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java]: not sure java doc is better if you want to comments the method.

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/RestAPIBuilder.java [hunk 2]
```diff
@@ -80,7 +88,9 @@ private String buildRequestUri() {
 
         if (params != null && params.size() > 0) {
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/RestAPIBuilder.java]: better to use `params.isEmpty()` ?

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java (hunk 1 above)
Comment (sophiaso) [reply spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java]: Updated Javadoc.

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/RestAPIBuilder.java (hunk 2 above)
Comment (sophiaso) [reply spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/RestAPIBuilder.java]: Updated.

**Pull request merged**